src/Utils/ArrayList.php

```diff
@@ -15,6 +15,8 @@
 /**
  * Provides the base class for a generic list (items can be accessed by index).
  * @template T
+ * @implements \IteratorAggregate<int, T>
```

Comment:
Isn't that duplicate information?

Reply:
It is, but just for getIterator(). Implementing IteratorAggregate also passes generic type to Traversable. Without it Traversable type is mixed. Not sure if this specific case is problem, but generally speaking, generic interface should be always implemented.

Generic type from getIterator() should be removed, as it is already defined by IteratorAggregate

Reply:
Thanks for the explanation. Can I ask @xificurk  to remove the annotation from `getIterator()`?